eppo_core/src/event_ingestion/event_ingestion.rs

```diff
@@ -0,0 +1,228 @@
+
+        let (input, flusher_uplink) = mpsc::channel(config.max_queue_size);
+        let (flusher_downlink, batcher_uplink) = mpsc::channel(1);
+        let (batcher_downlink, delivery_uplink) = mpsc::channel(1);
+        let (delivery_downlink, retry_uplink) = mpsc::channel(1);
+        let (delivery_status_tx, delivery_status_rx) = mpsc::channel(1);
+
+        runtime.spawn_untracked(auto_flusher::auto_flusher(
+            flusher_uplink,
+            flusher_downlink,
+            config.delivery_interval,
+        ));
+        runtime.spawn_untracked(batcher::batcher(
+            batcher_uplink,
+            batcher_downlink.clone(),
+            config.batch_size,
+        ));
+        runtime.spawn_untracked(delivery::delivery(
+            delivery_uplink,
+            delivery_downlink,
+            delivery_status_tx.clone(),
+            config.max_retries,
+            event_delivery,
+        ));
+        runtime.spawn_untracked(retry::retry(
+            retry_uplink,
+            // This is sending to input of delivery
+            batcher_downlink,
+            config.max_retries,
+            config.retry_interval,
+            config.max_retry_delay,
+        ));
+
+        // For now, nobody is interested in delivery statuses.
+        let _ = delivery_status_rx;
+
+        EventIngestion { tx: input }
+    }
+
+    pub fn track(&self, event_type: String, payload: serde_json::Value) {
+        let event = Event {
+            uuid: Uuid::new_v4(),
+            timestamp: crate::timestamp::now(),
+            event_type,
+            payload,
+        };
+
+        self.track_event(event);
+    }
+
+    fn track_event(&self, event: Event) {
+        let result = self
+            .tx
+            .try_send(BatchedMessage::singleton(QueuedEvent::new(event)));
+
+        if let Err(err) = result {
+            log::warn!(target: "eppo", "failed to submit event to event ingestion: {}", err);
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::event_ingestion::event::Event;
+    use crate::timestamp::now;
+    use serde_json::json;
+    use tokio::time::Duration;
+    use uuid::Uuid;
+    use wiremock::matchers::{body_json, header, method, path};
+    use wiremock::{Mock, MockServer, ResponseTemplate};
+
+    fn init() {
+        let _ = env_logger::try_init();
+    }
+
+    #[tokio::test]
+    async fn test_dispatch_starts_delivery() {
+        init();
+
+        let event = new_test_event();
+
+        let mock_server = MockServer::start().await;
+        Mock::given(method("POST"))
+            .and(path("/"))
+            .and(body_json(&json!({
+                "eppo_events": [event.clone()],
+            })))
+            .and(header("x-eppo-token", "test-sdk-key"))
+            .respond_with(ResponseTemplate::new(200).set_body_json(json!({
+                "failed_events": []
+            })))
+            .expect(1)
```

Comment:
The tests have been refactored to rely on wiremock's verification (simpler tests + nicer failure messages).